tools/compare_benchmark.py

```diff
@@ -0,0 +1,159 @@
+    baseline, _ = os.path.splitext(os.path.basename(baseline_out))
+    contender, _ = os.path.splitext(os.path.basename(contender_out))
+    comparison_out = os.path.join(out_dir, f"{baseline}_vs_{contender}.json")
+    subprocess.check_call(
+        f"third_party/benchmark/tools/compare.py -d {comparison_out} benchmarks {baseline_out} {contender_out}",
+        shell=True,
+    )
+    return comparison_out
+
+
+def load_comparison(comparison_out: str) -> list[Comparison]:
+    comparisons: list[Comparison] = []
+    with open(comparison_out) as f:
+        data = json.loads(f.read())
+
+    for row in data:
+        for measurement in row["measurements"]:
+            if row["run_type"] != "iteration":
+                continue
+            comparisons.append(
+                Comparison(
+                    row["name"],
+                    measurement["real_time"],
+                    measurement["real_time_other"],
+                    measurement["time"] + 1,
+                    row["time_unit"],
+                )
+            )
+
+    return comparisons
+
+
+def summarize_comparison(comparisons: Iterable[Comparison], out_dir: str) -> None:
+    sorted_comparisons = sorted(comparisons, key=lambda x: x.change)
+
+    # Print top improvements and regressions.
+    num_tops = 5
+    print(f"Top {num_tops} improvements:")
+    for i in range(min(num_tops, len(comparisons))):
+        comparison = comparisons[i]
+        if comparison.change >= 1:
+            break
+        print(f"  {comparison}")
+    print()
+    print(f"Top {num_tops} regressions:")
+    for i in range(min(num_tops, len(comparisons))):
+        comparison = comparisons[-(i + 1)]
+        if comparison.change <= 1:
+            break
+        print(f"  {comparison}")
+
+    # Generate and save the histogram of time changes.
+    plt.xlabel("Time change (contender/baseline)")
+    plt.ylabel("Number of benchmarks")
+    n, bins, patches = plt.hist([comparison.change for comparison in comparisons])
+    bin_centers = np.diff(bins) / 2 + bins[:-1]
+    for bar, count, x in zip(patches, n, bin_centers):
+        plt.text(x, count + 0.5, str(count), ha="center", va="bottom")
+
+    histogram_out = os.path.join(out_dir, "histogram.png")
+    plt.savefig(histogram_out)
+    print()
+    print(f"Saved the histogram of time changes to {histogram_out}.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("baseline", type=str, help="The baseline branch or commit")
+    parser.add_argument("contender", type=str, help="The contender branch or commit")
+    parser.add_argument(
+        "out_dir",
+        type=str,
+        help="The output folder that will contain benchmark results and comparison",
+    )
+    parser.add_argument(
+        "--benchmark_args",
+        type=str,
+        help="Arguments passed to nvfuser_bench, e.g., --benchmark_filter=NvFuserScheduler",
+    )
```

Comment:
You could use `parser.add_argument('benchmark_args', nargs=argparse.REMAINDER)` to avoid having to pack additional args into a string on the cmdline.